lib/BrowserStackTest.php

```diff
@@ -1,40 +1,40 @@
 <?php
+require_once('vendor/autoload.php');
+require_once('lib/globals.php');
 
-require 'vendor/autoload.php';
-require 'lib/globals.php';
+use Facebook\WebDriver\Remote\RemoteWebDriver;
 
-class BrowserStackTest extends PHPUnit\Framework\TestCase
-{
+class BrowserStackTest extends PHPUnit\Framework\TestCase {
     protected static $driver;
     protected static $bs_local;
 
-    public static function setUpBeforeClass()
-    {
+    public static function setUpBeforeClass(): void {
         $CONFIG = $GLOBALS['CONFIG'];
         $task_id = getenv('TASK_ID') ? getenv('TASK_ID') : 0;
 
-        $url = "https://" . $GLOBALS['BROWSERSTACK_USERNAME'] . ":" . $GLOBALS['BROWSERSTACK_ACCESS_KEY'] . "@" . $CONFIG['server'] ."/wd/hub";
-        $caps = $CONFIG['environments'][$task_id];
+        $url = "https://{$GLOBALS['BROWSERSTACK_USERNAME']}:{$GLOBALS['BROWSERSTACK_ACCESS_KEY']}@hub.browserstack.com/wd/hub";
+        $caps = $CONFIG['capabilities'][$task_id];
 
-        foreach ($CONFIG["capabilities"] as $key => $value) {
-            if(!array_key_exists($key, $caps))
-                $caps[$key] = $value;
-        }
-
-        if(array_key_exists("browserstack.local", $caps) && $caps["browserstack.local"])
+        if(array_key_exists("local", $caps["bstack:options"]) && $caps["bstack:options"])
         {
             $bs_local_args = array("key" => $GLOBALS['BROWSERSTACK_ACCESS_KEY']);
             self::$bs_local = new BrowserStack\Local();
+            print("Starting Local Binary...\n");
```

Comment:
Incorrect placement. This will start and stop binary for each test in parallel.php